core/src/engine/Hamiltonian_Heisenberg.cu

```diff
@@ -1,38 +1,37 @@
 #ifdef SPIRIT_USE_CUDA
```

Comment:
@MSallermann can you take a closer look at this file, and maybe also execute tests using CUDA on this branch?

These changes are quite difficult to review and make sure they're OK just from looking at them...